superset/views/base_api.py

```diff
@@ -145,7 +145,7 @@ def apply(self, query: Query, value: Any) -> Query:
             return query
         users_favorite_query = db.session.query(FavStar.obj_id).filter(
             and_(
-                FavStar.user_id == g.user.get_id(),
+                FavStar.user_id == get_user_id(),
```

Comment:
Mixed types, though likely benign from the database's perspective—at least for MySQL.